src/main/java/fr/moribus/imageonmap/commands/maptool/GiveCommand.java

```diff
@@ -36,109 +36,110 @@
-        }
-
-        switch (subs[0]) {
-            case "player":
-                return null;//temp
-            // return offlinePlayerParameter(subs[1]);
-
-            case "uuid":
-                StringBuffer string = new StringBuffer(subs[1].toLowerCase());
-                //if there are no '-'
-                if (string.length() == 32) {
-                    //we try to fix it by adding - at pos 8,12,16,20
-                    Integer[] pos = {20, 16, 12, 8};
-                    for (int i : pos) {
-                        string = string.insert(i, "-");
-                    }
-                }
-
-                //if the given uuid is well formed with 8-4-4-4-12 = 36 chars in length (including '-')
-                if (string.length() == 36) {
-                    return Bukkit.getOfflinePlayer(UUID.fromString(string.toString()));
-                }
-
-                throwInvalidArgument(
-                        I.t("Invalid uuid, please provide an uuid of this form xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
-                                + " or xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"));
-                break;
-            case "bank":
-                throwInvalidArgument(I.t("Not supported yet"));
-                break;
-
-            default:
-                throwInvalidArgument(I.t("Invalid prefix, valid one are: player | uuid"));
-        }
-        /*} catch (InterruptedException | ExecutionException e) {
-            PluginLogger.warning(I.t("Can't access to mojang API to check the player UUID"));
-        }*/
-        return null;
-    }
-
     @Override
     protected void run() throws CommandException {
 
         if (args.length < 2) {
             throwInvalidArgument(I.t("You must give a valid player name and a map name."));
         }
 
-        final Player p = getPlayerParameter(0);
-
-        ImageMap map;
-        //TODO add support for map name with spaces "cool name" or name or "name" "cool name with a \" and some stuff"
-        // should work
-        OfflinePlayer player = null;
+        ArrayList<String> arguments = getArgs();
 
-        if (args.length < 4) {
-            if (args.length == 2) {
-                player = playerSender();
+        if (arguments.size() > 3) {
+            warning(I.t("Too many parameters! Usage: /maptool give <playername> [playerFrom]:<mapname>"));
+            return;
+        }
+        if (arguments.size() < 1) {
+            warning(I.t("Too few parameters! Usage: /maptool give <playername> [playerFrom]:<mapname>"));
+            return;
+        }
+        final String mapName;
+        final String from;
+        final String playerName;
+        final Player playerSender;
+        Player playerSender1;
+        try {
+            playerSender1 = playerSender();
+        } catch (CommandException ignored) {
+            if (arguments.size() == 2) {
+                throwInvalidArgument(I.t("Usage: /maptool give <playername> <playerFrom>:<mapname>"));
```

Comment:
 `throwInvalidArgument` affiche la doc à l'utilisateur.
```suggestion
                throwInvalidArgument(I.t("Player name is required from the console"));
```